TESTS/mbed_hal/watchdog_reset/main.cpp

```diff
@@ -264,6 +266,48 @@ void test_kick_reset()
     TEST_ASSERT_MESSAGE(0, "Watchdog did not reset the device as expected.");
 }
 
+template<uint32_t window>
+void test_timeout()
+{
+    watchdog_features_t features = hal_watchdog_get_platform_features();
+    if (TIMEOUT_LOWER_LIMIT > features.max_timeout) {
+        TEST_IGNORE_MESSAGE("Requested timeout value not supported for this target -- ignoring test case.");
+        return;
+    }
+
+    // Phase 2. -- verify the test results.
+    if (current_case.received_data != CASE_DATA_INVALID) {
+        TEST_ASSERT_EQUAL(CASE_DATA_PHASE2_OK, current_case.received_data);
+        current_case.received_data = CASE_DATA_INVALID;
+        return;
+    }
+
+    // Phase 1. -- run the test code.
+    watchdog_config_t config = { TIMEOUT_LOWER_LIMIT };
+    TEST_ASSERT_EQUAL(WATCHDOG_STATUS_OK, hal_watchdog_init(&config));
+
+    // Kick watchdog before timeout.
+    // Watchdog should not trigger before timeout.
+    // If device restarts while waiting for the kick, test fails.
```

Comment:
Currently this produces an `ERROR` test case result instead of `FAIL`. Could you update that?